polyval/src/backends/clmul.rs

```diff
@@ -108,17 +61,72 @@ impl UniversalHash for Polyval {
     }
 
     /// Get GHASH output
-    fn finalize(self) -> Tag {
+    fn finalize(self) -> Output<Self> {
         unsafe { core::mem::transmute(self.y) }
     }
 }
 
 #[inline]
+#[target_feature(enable = "pclmulqdq")]
+#[target_feature(enable = "sse4.1")]
+unsafe fn mul(h: __m128i, x: __m128i) -> __m128i {
```

Comment:
I extracted this method when I was confused about a performance discrepancy (which turned out to be me passing `target-cpu` inconsistently).

Not sure if it's helpful or not.